[Feature] 회원가입 및 프로필 생성 기능 추가

dev-msj opened the pull request:
## 작업 내용

닉네임 중복 확인

회원가입
- UserAuth 생성

프로필 생성
- UserInfo 생성
- UserCategory 생성

## 이슈 번호
#34

dev-msj committed:
docs : 회원가입 기능 정의

dev-msj committed:
feat(UserAuthMapper) : 신규 UserAuth insert 기능 추가

dev-msj committed:
feat(UserAuthRepository) : creatUserAuth 메서드 추가

dev-msj committed:
refactor(UserAuth) : 외부 통신에 사용될 수 있도록 수정

AllArgsConstructor, NoArgsConstructor 추가
Serializable 상속

dev-msj committed:
feat(UserAuthService) : createUserAuth 메서드 추가

salt 생성, password hashing 후 UserAuth Insert

dev-msj committed:
feat(UserAuthController) : createUserAuth 메서드 추가

dev-msj committed:
Merge branch 'feature/feed-write' into feature/user-join

# Conflicts:
#	src/main/java/com/flab/oasis/service/UserAuthService.java

dev-msj committed:
refactor(UserAuthService) : createUserAuth에서 인증된 uid를 사용하도록 수정

dev-msj committed:
Merge branch 'fix/login-google/login-handler' into feature/user-join

# Conflicts:
#	src/main/java/com/flab/oasis/service/UserAuthService.java

dev-msj committed:
fix(UserAuthMapper) : 파라메터명 오류 수정

dev-msj committed:
Merge branch 'main' into feature/user-join

# Conflicts:
#	src/main/java/com/flab/oasis/controller/UserAuthController.java
#	src/main/java/com/flab/oasis/model/UserAuth.java
#	src/main/java/com/flab/oasis/service/UserAuthService.java

dev-msj committed:
Merge branch 'main' into feature/user-join

dev-msj committed:
fix(UserAuthService) : merge된 내용 반영

dev-msj committed:
Merge branch 'main' into feature/user-join

dev-msj committed:
refactor(UserAuthService) : createUserAuth 로직 수정

UserRole을 'USER'로 설정
socialYN이 'N'일 때 비밀번호를 hashing하도록 변경
LoginResult를 반환하도록 수정

dev-msj committed:
refacot(UserAuthController) : 회원가입 시 LoginResult를 반환하도록 수정

dev-msj committed:
docs : 회원가입 절차 재정의

dev-msj committed:
feat : UserInfo 모델 추가

dev-msj committed:
fix(UserAuthMapper) : tag명 오류 수정

insert 구문의 tag명 수정 select -> insert

dev-msj committed:
refactor(UserAuthMapper) : UserAuth 생성시 UserRole도 생성하도록 수정

dev-msj committed:
refactor(UserCategory) : Builder 패턴 추가

dev-msj committed:
feat(UserCategoryMapper) : 카테고리 생성 기능 추가

dev-msj committed:
feat : UserProfile 모델 추가

UserInfo와 UserCategory를 통합하여 client와 통신하는 객체

dev-msj committed:
feat(UserInfoMapper) : 닉네임 중복 확인 쿼리 추가

dev-msj committed:
feat : 닉네임 중복확인 기능 추가

dev-msj committed:
refactor(MyBatisConfigUser) : transaction manager 추가

dev-msj committed:
feat(UserInfoMapper) : USER_INFO 생성 및 조회 쿼리 추가

dev-msj committed:
feat : NotFoundException 추가

dev-msj committed:
feat(UserInfoRepository) : UserInfo 생성 및 조회 기능 추가

UserInfo 조회 시 redis에서 먼저 조회.
조회 결과가 null이면 DB에서 가져온 후 redis에 저장.

DB에서 조회 시 null일 경우 NoFoundException을 던지도록 함.
-> NotFoundException일 경우 client에서 프로필 생성 화면을 보여주게 하기 위함

dev-msj committed:
feat(UserCategoryRepository) : UserCategory 생성 및 조회 기능 추가

UserCategory 조회 시 redis에서 먼저 조회.
조회 결과가 null이면 DB에서 가져온 후 redis에 저장.

dev-msj committed:
feat(UserProfileService) : User Profile을 생성 및 조회하는 기능 추가

User Profile을 생성할 때 Transactional 선언으로 에러 시 rollback되도록 처리

dev-msj committed:
feat(UserInfoController) : User Profile 생성 및 조회 API 추가

dev-msj committed:
feat(ExceptionAdvice) : NotFoundException handler 추가

dev-msj committed:
fix(UserCategoryMapper) : UserCategoryList를 foreach 구문으로 처리

dev-msj committed:
fix(UserProfileService) : UserCategory 생성 관련 기능 오류 수정

uid를 SecurityContext에서 받아오도록 수정
UserCategory가 empty가 아닐 경우만 insert하도록 수정

dev-msj committed:
refactor(BookCategory) : Getter를 Lombok 어노테이션으로 대체

dev-msj committed:
refactor(HomeService) : UserCategoryRepository 반영

기존 UserCategoryMapper를 UserCategoryRepository로 대체

jinspark-lab commented on src/main/java/com/flab/oasis/controller/UserProfileController.java:
```diff
@@ -0,0 +1,30 @@
+    public boolean createUserProfile(@RequestBody UserProfile userProfile) {
+        userProfileService.createUserProfile(userProfile);
+
+        return true;
```
응답이 항상 true 를 반환하는 이유가 있나요?

dev-msj replied:
create 같은 경우에는 로직이 단순히 insert 구문을 실행하는 것이기 때문에 정상적으로 쿼리가 동작했으면 true를 반환하게 했습니다.
만약 쿼리에 필요한 parameter 값들 중 이상이 있다면 ExceptionAdvice가 SQLException을 캐치하여 메세지를 전달하도록 되어있습니다.

하지만 항상 반환하는 true를 구조가 client에게, 그리고 해당 코드를 보는 사람에게 혼동을 줄 수 있을 수도 있다는 생각이 드네요...
쿼리 실행 결과를 response 모델을 작성한 후 해당 객체에 실행 결과(boolean)와 메세지를 반환하는 형태가 좋을 것 같다는 생각이 듭니다.

dev-msj replied:
위 말씀드린 내용 반영했습니다.
변경된 내용에 맞게 Wiki의 API 문서도 수정하도록 하겠습니다.

jinspark-lab replied:
잘못이해하신 것 같아서 피드백 다시 드렸습니다.

jinspark-lab commented on src/main/java/com/flab/oasis/repository/UserCategoryRepository.java:
```diff
@@ -0,0 +1,50 @@
+
+    private List<UserCategory> getUserCategoryByUidFromDB(String uid) {
+        return Optional.ofNullable(userCategoryMapper.getUserCategoryByUid(uid))
+                .orElse(new ArrayList<>());
```
이 부분이 좀 어색한데, getUserCategoryByUid 는 List 를 반환하지 않나요? null 이 반환되는 경우가 있을까요?

dev-msj replied:
List형을 반환하니 empty 상태인 List 객체를 반환하겠네요...
해당 내용 수정했습니다.

또한 User Category List를 반환하니 반환형에 맞게 메서드명 또한 getUserCategoryListByUid로 수정했습니다.

jinspark-lab commented on src/main/java/com/flab/oasis/repository/UserCategoryRepository.java:
```diff
@@ -0,0 +1,50 @@
+
+            return userCategoryList;
+        } catch (Exception e) {
+            return getUserCategoryByUidFromDB(uid);
```
Redis 에 없으면 DB 를 조회해서 가져오는 로직인가요? try catch 로 처리하시는 이유가 뭘까요?

dev-msj replied:
Redis에 문제가 생겨 접속이 안될 경우에는 try-catch 구문을 통해 DB에서 직접 가져올 수 있도록 했습니다.

jinspark-lab reviewed (commented):
1차 피드백드렸습니다

dev-msj committed:
refactor(UserCategoryRepository) : 무의미한 코드 수정

기본적으로 List형을 반환하므로 Nullable로 List 객체를 생성하는 코드 제거

dev-msj committed:
style : 유저 카테고리를 가져오는 메서드명 수정

UserCategory List를 반환하므로 반환형에 맞게 메서드명 수정
getUserCategoryByUid -> getUserCategoryListByUid

dev-msj committed:
feat(SQLResultResponse) : SQL 실행 결과를 반환하는 객체 추가

dev-msj committed:
refactor : 프로필 생성 관련 쿼리 excpetion 처리

spring은 Runtime 중 발생하는 모든 SQLException을 DataAccessException으로 포장해서 던진다.
그러므로 Repository에서 DataAccessException으로 catch하여 다시 SQLException를 throw하도록 변경

dev-msj committed:
refactor : 프로필 생성 결과를 SQLResultResponse로 반환하도록 수정

jinspark-lab commented on src/main/java/com/flab/oasis/controller/UserProfileController.java:
```diff
@@ -17,10 +18,8 @@ public boolean isExistsNickname(@PathVariable String nickname) {
-
-        return true;
+    public SQLResultResponse createUserProfile(@RequestBody UserProfile userProfile) {
+        return userProfileService.createUserProfile(userProfile);
```
Wrapping 하신 건 좋은 선택입니다만, 그런 의미는 아니었습니다. userProfileService.createUserProfile 은 void 를 반환하는 함수이고, 결국에 이 동작은 true 또는 exception 을 반환합니다. 퀄리티와 별개로 exception 을 일종의 응답처럼 쓰실거면 차라리 Return 을 안해도 됩니다. 그 관점에서 이런 API 디자인은 모순이 있습니다. 한번 더 고민해보시는게 좋을 것 같습니다.

dev-msj replied:
API 디자인 형태에 대해 다시 고민해보니 client에게 일반화된 형태의 response를 제공해주는 것이 좋을 것 같다는 생각이 들었습니다.
그래서 다음과 같이 일반화된 형식을 가진 GeneralResponse 모델을 추가하였습니다.

- code : 0(성공), 0이 아닌 번호(에러 코드)
- message : 응답 또는 에러 메세지
- data : 요청된 자원에 대한 결과값(generic)

이 형식을 활용해 모든 request에 대하여 일반화된 형식을 가진 response를 반환할 수 있도록 하려고 합니다.

jinspark-lab commented on src/main/java/com/flab/oasis/repository/UserCategoryRepository.java:
```diff
@@ -0,0 +1,49 @@
+
+            return userCategoryList;
+        } catch (Exception e) {
+            return userCategoryMapper.getUserCategoryListByUid(uid);
```
이 부분 역시 논리적으로 올바르지 않습니다. 지적할 부분이 좀 많은데

1. Exception 클래스가 Redis 에 접속이 안된다는것과 동치가 아닙니다.
2. try catch 를 통해 System Fail handling 을 할만한 요소가 아닙니다.
3. DB 를 갖고오는 로직에서 Exception 이 발생해도 DB 에 쿼리를 날리나요?

dev-msj replied:
1. 말씀해주신 대로 Exception 클래스를 catch하는 것은 잘못된 구조가 맞네요...DB에서 문제가 발생해도 Exception이 잡아버리기 때문에 이러한 예외처리는 잘못됐습니다. 그래서 RedisConnectionFailureException으로 대체하여 Redis 연결 실패 시에만 동작할 수 있도록 수정했습니다.
2. redis의 경우 캐싱이 목적이기 때문에 시스템 운영 자체가 불가능한 요소가 아니라고 생각했습니다. 그래서 DB가 동작하는 한 성능이 떨어지더라도 서비스는 제공할 수 있어야 하기 때문에 try-catch 구문을 활용하였습니다. 하지만 이 경우 또한 System Fail에 관한 요소에 해당하니 try-catch가 아닌 다른 방법을 찾아봐야겠네요...
3. DB 관련 Exception이 발생해도 catch하여 다시 DB에 쿼리를 날리게 되는 구조네요...1번 답변과 중복되는 내용으로 RedisConnectionFailureException으로 변경하여 redis 연결 실패에 대한 예외를 처리하도록 수정하였습니다. DB과 관련된 Exception은 ExceptionAdvice가 DataAccessException을 catch하여 client에게 Internal Servcer Error를 응답하도록 하였습니다.

dev-msj replied:
추가적으로 Optional에 관한 부분도 삼항연산자를 통해 null처리를 하도록 변경했습니다.
이상하게 redis에서 가져온 값이 존재함에도 불구하고 orElse(혹은 orElseThrow) 메서드를 탑니다.
현재는 무슨 이유인지 예상되는 바가 없는데, 우선 의도한 목적대로 로직이 수행될 수 있도록 수정했습니다.

jinspark-lab reviewed (changes requested):
대부분 같은 이슈 반복이어서 중복 피드백은 안드렸습니다. 이번에는 논리적으로 틀린 코드들이 많네요. 확인 부탁드립니다.

dev-msj committed:
feat(ResultResponse) : API 응답용 공통 모델 추가

Generic을 통해 응답할 data의 클래스를 추상화

dev-msj committed:
refactor : UserInfo/Category repository에서 try-catch 삭제

dev-msj committed:
refactor(UserProfileService) : ExceptionAdvice가 handling하도록 수정

SQLException은 모든 Service의 공통 관심사이므로 ExceptionAdvice가 처리하도록 수정

dev-msj committed:
refactor(UserProfileService) : ResultResponse를 반환하도록 수정

dev-msj committed:
refctor(UserProfileService) : NotFoundException을 catch하도록 수정

dev-msj committed:
refactor(UserProfileController) : ResultResponse를 응답하도록 수정

dev-msj committed:
refactor(AuthenticationException) : field에 value 추가

dev-msj committed:
style : 사용되지 않는 모델 삭제

dev-msj committed:
refactor(ExceptionAdvice) : ResultResponse로 응답하도록 수정

dev-msj committed:
refactor(JwtFilter) : 토큰 갱신 응답을 ResultResponse로 대체

토큰 갱신 응답을 ResponseReissuedJWT를 ResultResponse로 대체
ResponseReissuedJWT 삭제

dev-msj committed:
refactor(UserProfileService) : 정상처리 시 code를 0으로 반환하도록 수정

dev-msj committed:
refactor : API Response 모델명 변경

모델명 변경 ResultResponse -> GeneralResponse

dev-msj committed:
docs(UserAuthService) : 유저 생성 로그 추가

dev-msj committed:
refactor : GeneralResponse를 Controller가 생성하도록 수정

dev-msj committed:
fix(UserInfoRepository) : redis 관련 로직 에러 수정

redis에 값이 없을 경우에만 db에서 가져온 값을 redis에 등록하도록 변경
redis의 get 결과를 직접 optional로 감쌀 경우 null로 처리되는 문제 해결
json convertValue 시 TypeReference가 아닌 class 정보를 직접 전달하도록 변경

dev-msj committed:
fix(UserInfoRepository) : redis 관련 exception 처리 변경

redis에 연결되지 않을 경우에만 DB를 조회하도록 Exception 변경
Exception -> RedisConnectionFailureException

dev-msj committed:
docs(UserInfoRepository) : redis 연결 실패 로그 추가

dev-msj committed:
refactor(ExceptionAdvice) : SQLException을 DataAccessException으로 대체

runtime 중에 발생한 jdbc에러를 DataAccessException로 감싸서 던지기 때문에 advice 대상을 변경

dev-msj committed:
Merge branch 'main' into feature/user-join

dev-msj committed:
style(ExceptionAdvice) : 사용되지 않는 import 제거

dev-msj committed:
Merge branch 'feature/user-join' of https://github.com/dev-msj/Oasis into feature/user-join

jinspark-lab commented on src/main/java/com/flab/oasis/controller/UserProfileController.java:
```diff
@@ -14,17 +14,35 @@ public class UserProfileController {
-        return userProfileService.isExistsNickname(nickname);
+    public GeneralResponse<Boolean> isExistsNickname(@PathVariable String nickname) {
+        return GeneralResponse.<Boolean>builder()
+                .code(0)
```
코드를 0으로 던지는건 별로 좋지않습니다. 0은 Integer 기본값이고, 이런건 RESPONSE CODE 를 명시해주는게 더 좋습니다

jinspark-lab commented on src/main/java/com/flab/oasis/controller/advice/ExceptionAdvice.java:
```diff
@@ -1,38 +1,37 @@
-    @ExceptionHandler(SQLException.class)
-    public ResponseEntity<ResultResponse<String>> handleSQLException(SQLException e) {
+    @ExceptionHandler(DataAccessException.class)
+    public ResponseEntity<GeneralResponse<String>> handleDataAccessException(DataAccessException e) {
```
ExceptionAdvice 에서 모든 예외를 보통 처리하지는 않습니다. 그럼 너무 무거워지고, 클라이언트가 모든 예외를 구별할 필요는 없으니까요

jinspark-lab commented on src/main/java/com/flab/oasis/filter/JwtFilter.java:
```diff
@@ -63,12 +63,13 @@ public void doFilterInternal(HttpServletRequest request, HttpServletResponse res
-                        .build();
-                String responseBody = new ObjectMapper().writeValueAsString(responseReissuedJWT);
+                GeneralResponse<JsonWebToken> generalResponse = GeneralResponse.<JsonWebToken>builder()
+                            .code(ErrorCode.RESET_CONTENT.getCode())
```
Access Code Expire 는 Error 라고 보기는 좀 애매합니다.

jinspark-lab commented on src/main/java/com/flab/oasis/model/GeneralResponse.java:
```diff
@@ -9,7 +9,7 @@
 @AllArgsConstructor
 @Builder
-public class ResultResponse<E extends Serializable> implements Serializable {
+public class GeneralResponse<E extends Serializable> implements Serializable {
```
Serializable 을 상속해서 Generic 으로 핸들링하고 계신데 좀 포괄적이고 따로 클래스 만들어서 좁혀주시면 더 좋습니다. (e.g. BaseResponse)

jinspark-lab reviewed (commented):
피드백드렸습니다.

jinspark-lab commented:
Redis & DataSource Configuiration 쪽 추상화 시간나실때 공부해보세요

dev-msj committed:
feat(ResponseCode) : 정상 응답용 코드 추가

dev-msj committed:
style(ErrorCode) : RESET_CONTENT 삭제

ResponseCode로 대체

dev-msj committed:
refactor(JwtFilter) : ResponseCode의 RESET_CONTENT로 대체

dev-msj committed:
refactor(UserProfileController) : 응답 코드에 ResponseCode 적용

dev-msj committed:
feat : BaseResponse 추가

dev-msj committed:
feat : SuccessResponse 추가

dev-msj committed:
refactor(UserProfileController) : isExistsNickname의 반환형을 boolean으로 변경

dev-msj committed:
refactor(UserProfileController) : createUserProfile의 결과로 SuccessResponse 반환

dev-msj committed:
feat : UserProfileResponse 추가

dev-msj committed:
refactor(UserProfileController) : getUserProfileByUid를 반환형 변경

GeneralResponse에서 UserProfileResponse로 대체

dev-msj committed:
refactor(ExceptionAdvice) : DataAccessException handler 삭제

dev-msj committed:
feat : FailureResponse 추가

dev-msj committed:
refactor(ExceptionAdvice) : 응답을 FailureResponse로 대체

dev-msj committed:
refactor(ExceptionAdvice) : exception이 발생한 value를 알려주지 않도록 변경

dev-msj committed:
feat : JsonWebTokenResponse 추가

dev-msj committed:
refactor(JwtFilter) : 토큰 갱신용 응답 모델 변경

GeneralResponse -> JsonWebTokenResponse

dev-msj committed:
style(GeneralResponse) : 사용되지 않는 모델 삭제

dev-msj committed:
style(UserProfileService) : 사용되지 않는 import 삭제

dev-msj committed:
fix(JwtServiceTest) : assert Exception 오류 해결

AuthorizationException을 삭제하고 모두 AuthenticationException으로 대체하였으나
미반영된 내용 해결

dev-msj committed:
refactor(FeedController) : SuccessResponse 적용

dev-msj committed:
feat : RedisErrorHandler 추가

Redis에 문제가 발생할 경우 서비스 오류가 발생하지 않고, Log 생성 후 DB의 값을 전달

dev-msj committed:
refactor(UserInfoRepository) : redisTemplate을 Cacheable로 대체

dev-msj committed:
style(RedisConfig) : 메서드에서 static 제거

dev-msj committed:
style(UserLoginRequest) : 불필요한 상속 제거

dev-msj committed:
fix(RedisConfig) : ClassCastException 문제 해결

redis의 data를 deserialize할 때 LinkedHashMap이 원하는 Model로 cast되지 않는 문제 해결
- 원인 : LinkedHashMap은 bootstrap loader에 의해 올라가고, Model은 app loader에 의해 올라가기 때문에
변환될 수 없음.
- 해결 : 직접 선언했던 ObjectMapper를 삭제하고, GenericJackson2JsonRedisSerializer에서 기본으로 생성하는
ObjectMapper를 사용하게 함으로써 해결됨.

dev-msj committed:
refactor : cacheable 어노테이션 적용

jinspark-lab commented on src/main/java/com/flab/oasis/model/response/JsonWebTokenResponse.java:
```diff
@@ -0,0 +1,17 @@
+@Setter
+@SuperBuilder
+public class JsonWebTokenResponse extends BaseResponse implements Serializable {
+    private static final long serialVersionUID = -4159196522419234339L;
```
BaseResponse 에 Serializable 이 달려있으므로 굳이 Serializable 은 Impement 안하셔도 됩니다.

dev-msj replied:
네 해당 내용 반영했습니다.

jinspark-lab commented on src/main/java/com/flab/oasis/model/response/FailureResponse.java:
```diff
@@ -0,0 +1,10 @@
+
+@SuperBuilder
+public class FailureResponse extends BaseResponse implements Serializable {
+    private static final long serialVersionUID = 7291705160666614470L;
```
빈 Interface 를 만드신다면, Validation Check 매커니즘을 넣는 것도 방법입니다. 
(예시)

`
public int getCode() {
    if (code == 200) {
        // Throw Exception
    }
}
`

dev-msj replied:
현재 FailureResponse를 사용할 때는 Builder 패턴을 활용하여 객체가 생성되면 바로 response를 하도록 되어있습니다.
그래서 Getter를 활용하는 것보다는 해당 객체가 생성될 때 code값이 validate 되는 것이 더 적합하다고 생각했습니다.
결과적으로 hiddenBuilder를 활용하여 code값을 받을 때 값이 200이면 UnexpectedCodeException을 throw하도록 수정하였습니다.

jinspark-lab commented on src/main/java/com/flab/oasis/controller/UserProfileController.java:
```diff
@@ -0,0 +1,44 @@
+            return UserProfileResponse.builder()
+                    .code(e.getErrorCode().getCode())
+                    .message(e.getMessage())
+                    .build();
```
컨트롤러에서 Exception 체크를 구현하시면 Controller 가 무거워지고 코드 관리가 어려워집니다.

dev-msj replied:
네 해당 로직을 UserProfileService에서 처리하도록 변경했습니다!

jinspark-lab commented on src/main/java/com/flab/oasis/service/UserAuthService.java:
```diff
@@ -36,6 +39,30 @@ public class UserAuthService {
+
+        return LoginResult.builder()
+                .jsonWebToken(
+                        jwtService.createJwt(userAuth.getUid(), userAuth.getUserRole())
```
Advanced 를 원하신다면 이 로직의 문제를 생각해보시는게 좋습니다.

dev-msj replied:
해당 로직의 문제점은 회원가입 메서드에 대한 결과를 반환하지 않고, Login 결과에 해당하는 LoginResult 반환한다는 점인 것 같습니다.
그래서 회원가입에 대한 결과로 JsonWebToken을 반환하도록 변경했습니다.
client 입장에서는 회원가입 요청에 대한 응답으로 토큰을 발급받았다면, 회원가입이 정상적으로 이뤄졌다고 판단할 수 있기때문입니다.

jinspark-lab reviewed (commented):
Redis 부분 빼고 피드백드립니다.

jinspark-lab commented:
Redis 테스트 코드 반영해주시면 감사하겠습니다

dev-msj committed:
refactor : Exception 로직 이전

UserProfileController
- getUserProfileByUid 메서드에서 NotFoundException catch 로직 삭제

UserProfileService
- getUserProfileByUid 메서드에 NotFoundException catch 로직 추가

dev-msj committed:
style(BaseResponse) : 중복된 implements 제거

dev-msj committed:
refactor : 회원가입 시 JWT만 반환하도록 변경

dev-msj committed:
feat(ErrorCode) : EXPECTATION_FAILED 코드 추가

dev-msj committed:
feat : UnexpectedCodeException 추가

dev-msj committed:
refactor(FailureResponse) : code 검증 로직 추가

hiddenBuilder를 통해 code가 200일 경우 UnexpectationCodeException을 던지도록 변경

dev-msj committed:
style : 사용하지 않는 import 제거

dev-msj committed:
refactor(TestMapper) : 범용 메서드를 사용하도록 변경

Select 어노테이션을 활용한 getTestData 메서드 추가
불필요한 Repository 어노테이션 삭제
cacheTest 메서드 삭제
dbConnectionTest 메서드 삭제

dev-msj committed:
style(TestMapper.xml) : 사용되지 않는 파일 삭제

dev-msj committed:
refactor(TestService) : Cache 기능만 Test하도록 변경

dbConnectionTest 메서드 삭제
EhCache/Redis의 Caching 및 Evict Test 메서드 추가
불필요한 import 삭제

dev-msj committed:
test(CommonTest) : MyBatis/Caching Test 재작성

MyBatis Connection Test
- 불필요하게 TestModel을 사용하지 않고 직접 데이터를 받아와 assert하도록 변경

Redis/EhCache Caching Test
- Mokito 적용
- 동일 메서드를 중복 호출하여 캐싱에 의해 실제 동작 횟수가 1회인지 확인

Redis/EhCache Cache Evict Test
- Mokito 적용
- 특정 메서드에 대한 캐시를 삭제한 후에 해당 메서드가 재호출되는지 확인

dev-msj committed:
style : 사용되지 않는 TestModel 삭제

jinspark-lab reviewed (commented):
확인했습니다. 잘 반영된 것 같습니다.
